### PR TITLE
fix: Handle missing header in merge commit

### LINKED ISSUE #691: Parser failing with TypeError: Cannot read property 'trim' of undefined
Parser fails with the following parameters:

```javascript
const { sync } = require("conventional-commits-parser");

console.log(
  sync("Merge pull request #134 from mgmeyers/syntax-decor-flag", {
    mergePattern: "^Merge pull request #(.*) from (.*)$",
    mergeCorrespondence: ["issueId", "source"],
  })
);
```

```
⇣6% ❯ node index.js
.../node_modules/conventional-commits-parser/lib/parser.js:152
    while (!header.trim()) {
                   ^

TypeError: Cannot read property 'trim' of undefined
    at parser (.../node_modules/conventional-commits-parser/lib/parser.js:152:20)
    at sync (.../node_modules/conventional-commits-parser/index.js:97:10)
    at Object.<anonymous> (/index.js:7:13)
    at Module._compile (internal/modules/cjs/loader.js:1200:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1220:10)
    at Module.load (internal/modules/cjs/loader.js:1049:32)
    at Function.Module._load (internal/modules/cjs/loader.js:937:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:71:12)
    at internal/main/run_main_module.js:17:47
```

Related problem https://github.com/marvinpinto/actions/issues/20#issuecomment-719964192.

### PR DESCRIPTION
**Pull request by eyal0**
This fixes #691

### CODE REVIEW

**Review by Vadorequest — APPROVED** [review]
LGTM

**Comment by eyal0** [comment]
I'm trying to test this but I don't know how.  When I use `npm test` on my computer, it doesn't work.

**Review by bcoe — APPROVED** [review]
Thank you for the fix 👍

File: packages/conventional-commits-parser/lib/parser.js [hunk 1]
```diff
@@ -147,9 +147,12 @@ function parser (raw, options, regex) {
     merge = mergeMatch[0]
 
     header = lines.shift()
-    while (!header.trim()) {
```
Comment (whogonestopme662901) [inline packages/conventional-commits-parser/lib/parser.js:150]: litecoin:

**Pull request merged**